### PR TITLE
[MODULAR] Aurora Station Knitting Port

### PR DESCRIPTION
**Pull request by Icarus-The-Sun**
## About The Pull Request
Modular-Adds Knitting from AuroraStation, alongside a new knitting skill that will speed up how fast someone can knit. Tested in game to function. Knit clothing!

HOW TO: To knit, simply put a ball of yarn onto a pair of knitting needles, use them in hand, and select the clothing you'd like to knit! Moving during this will stop the process. The color of the ball of yarn will affect the color of the clothing you produce from the radial menu shown (similar to picking a module and form when playing a borg). Finishing knitting clothing will gain knitting skill experience and gaining skill levels will let you knit faster. The default speed is two minutes for clothing item at this time. You can get yarn and needles from cargo, or from the loadout menu in the toys section.

(Code PRs for knitting on Aurora are https://github.com/Aurorastation/Aurora.3/pull/13114, https://github.com/Aurorastation/Aurora.3/pull/11932, https://github.com/Aurorastation/Aurora.3/pull/4749, and https://github.com/Aurorastation/Aurora.3/pull/12179)

## Changelog

:cl:
add: Adds the knitting system from Aurora.
add: Adds knitting supplies to the Toy section of the loadout menu.
add: Adds a knitting crate to cargo's orders!
add: Adds a knitting skill.
/:cl:

### CODE REVIEW

**Comment by Icarus-The-Sun** [comment]
I am aware that scarves have errors in the radial menu, the code is being a bitch.

File: modular_zubbers/code/modules/knitting/knitting.dm [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	/obj/item/clothing/suit/costume/ianshirt,
+	/obj/item/clothing/suit/toggle/jacket/sweater,
+	/obj/item/clothing/suit/costume/ghost_sheet,
+	/obj/item/clothing/neck/scarf/knitted,
+	/obj/item/clothing/head/beanie/knitted,
+	/obj/item/clothing/gloves/color/grey/protects_cold,
+	/obj/item/clothing/suit/hooded/wintercoat/skyrat,
+	/obj/item/clothing/neck/mantle,
+	/obj/item/clothing/accessory/armband/knitted,
+	/obj/item/clothing/under/misc/pj/red,)//Add more? Get Radial thumbnails fixed.
```
Comment (BurgerLUA) [inline modular_zubbers/code/modules/knitting/knitting.dm:31]: This list should be formatted better.

Comment (BurgerLUA) [reply modular_zubbers/code/modules/knitting/knitting.dm:31]: Actually this should probably also be a global list.

File: modular_zubbers/code/modules/knitting/knitting.dm [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	var/static/list/name2knit
```
Comment (BurgerLUA) [inline modular_zubbers/code/modules/knitting/knitting.dm:32]: A static list does not work here like this. I think instead you should make this a global instead.

**Review by BurgerLUA — CHANGES_REQUESTED** [review]

File: modular_zubbers/code/modules/knitting/knitting.dm (hunk 2 above)
Comment (Icarus-The-Sun) [reply modular_zubbers/code/modules/knitting/knitting.dm]: To clarify, is this referring to name2knit here?

Comment (BurgerLUA) [reply modular_zubbers/code/modules/knitting/knitting.dm]: Yes, name2knit.

**Comment by Icarus-The-Sun** [comment]
Clothing icons have partially gotten fucked up, will be fixed. Implemented a skill that will let people knit faster the more they do it, base time for knitting at 2 minutes.

**Comment by Icarus-The-Sun** [comment]
All of the knitting PR is working without any errors as of now, tested in game! The knittable cloak looks odd when knitted with anything but white yarn, but that's a minor spriting issue and can be quickly fixed.

**Comment by PastelPrinceDan** [comment]
since this is still only tm'd i'll just report issues here:
- bunch of items are broken. knitted sweaters, beanies and scarves get you this bad boy just below when you try to knit them, in any color. every other item seems to work.
image
- maxing out the skill and getting the legendary knitting cloak works, BUT you can't pick up the cloak despite having the skill maxed out

...on the topic of maxing out the skill, making it easier to achieve would be nice. it's very slow and tedious as it is now, and takes about an hour and a half. i wasted my time on it to see the reward, but i know no one else would because GOD is it boring

**Comment by Icarus-The-Sun** [comment]
> since this is still only tm'd i'll just report issues here:
> 
> * bunch of items are broken. knitted sweaters, beanies and scarves get you this bad boy just below when you try to knit them, in any color. every other item seems to work.
>   image
> * maxing out the skill and getting the legendary knitting cloak works, BUT you can't pick up the cloak despite having the skill maxed out
> 
> ...on the topic of maxing out the skill, making it easier to achieve would be nice. it's very slow and tedious as it is now, and takes about an hour and a half. i wasted my time on it to see the reward, but i know no one else would because GOD is it boring

Right, I'll have to update the sprites for that then, and possibly double the exp gain, or more.

**Comment by nevimer** [comment]
I don't want this to be thrown out, I think it's pretty good, and I want to merge it and it requires you to fix your branch to use the latest master.  I should have merged this before rebasing.

**Comment by StrangeWeirdKitten** [comment]
Closing due to it being stale and having conflicts.

**Pull request closed without merging**